tests/test_pyav.py

```diff
@@ -0,0 +1,329 @@
+from imageio.plugins.pyav import _format_to_dtype  # noqa: E402
+
+
+def test_mp4_read(test_images: Path):
+    with av.open(str(test_images / "cockatoo.mp4"), "r") as container:
+        for idx, frame in enumerate(container.decode(video=0)):
+            if idx == 4:
+                break
+
+    # ImageIO serves the data channel-first
+    expected = frame.to_ndarray(format="rgb24")
+
+    result = iio.imread(
+        test_images / "cockatoo.mp4", index=42, plugin="pyav", format="rgb24"
+    )
+    np.allclose(result, expected)
+
+    result = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=42,
+        plugin="pyav",
+        constant_framerate=False,
+        format="rgb24",
+    )
+    np.allclose(result, expected)
+
+
+def test_mp4_writing(tmp_path, test_images):
+    frames = iio.imread(test_images / "newtonscradle.gif", index=None)
+
+    with av.open(str(tmp_path / "expected.mp4"), mode="w") as container:
+        stream = container.add_stream("libx264", rate=None)
+        stream.width = frames.shape[2]
+        stream.height = frames.shape[1]
+
+        for frame in frames:
+            av_frame = av.VideoFrame.from_ndarray(frame, format="rgb24")
+            for packet in stream.encode(av_frame):
+                container.mux(packet)
+
+        for packet in stream.encode():
+            container.mux(packet)
+
+    mp4_bytes = iio.imwrite(
+        "<bytes>",
+        frames,
+        format_hint=".mp4",
+        plugin="pyav",
+        codec="libx264",
+    )
+
+    expected = Path(tmp_path / "expected.mp4").read_bytes()
+    # actual = Path(tmp_path / "actual.mp4").read_bytes()
+    assert expected == mp4_bytes
+
+
+def test_metadata(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        meta = plugin.metadata()
+        assert meta["profile"] == "High 4:4:4 Predictive"
+        assert meta["codec"] == "h264"
+
+        plugin.metadata(index=4)
+    print("")
+
+
+def test_properties(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        with pytest.raises(IOError):
+            # subsampled format
+            plugin.properties(format="yuv420p")
+
+        with pytest.raises(IOError):
+            # components per channel differs
+            plugin.properties(format="nv24")
+
+        props = plugin.properties()
+        assert props.shape == (3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is False
+
+        props = plugin.properties(index=4, format="rgb48")
+        assert props.shape == (720, 1280, 3)
+        assert props.dtype == np.uint16
+        assert props.is_batch is False
+
+        props = plugin.properties(index=None)
+        assert props.shape == (280, 3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is True
+
+
+def test_video_format_to_dtype():
+    fake_formats = [
```

Comment:
I'm curious why these are fake formats. There are presumably many fake formats. Why choose these?

Reply:
Maybe `fake_formats` is not a good name for them. All of these make sense, and they are typically related to hardware acceleration (which is currently unsupported in pyAV).

The reason I call them fake formats here is because we can instantiate them as a format (`av.VideoFormat(fake_format_name)` succeeds), but these formats have no components, so we can't deduce an appropriate dtype from these so we have to raise instead.

Reply:
could you add a comment in the code to that effect
```suggestion
    # All of these make sense, and they are typically related to hardware acceleration,
    # which is  unsupported in pyAV at the time of writing (2022/03/06)

    # We can instantiate them as a format (av.VideoFormat(fake_format_name) succeeds)
    # but these formats have no components. Therefore we can't deduce an appropriate 
    # dtype from these so we have to raise instead.
    fake_formats = [
```

Reply:
done.